### PR TITLE
feat(Custom Field): add support to custom field with template

### PR DESCRIPTION
**Pull request by edetec**

### CODE REVIEW

**Review by lincrosenbach — APPROVED** [review]

File: src/ng-xform/field-components/base-dynamic-field.component.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export class BaseDynamicFieldComponent<T extends DynamicField> implements OnInit
   control: FormControl;
   visible = true;
   public hideLabelOnEdit = false;
+  public useFormattedValueOnReadonly = true;
```
Comment (lvoliveira) [inline src/ng-xform/field-components/base-dynamic-field.component.ts]: Could you add some docs for this attribute? What is the formatted value?

File: src/ng-xform/fields/custom-field.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { TemplateRef } from '@angular/core';
+import { DynamicField } from './dynamic-field';
+
+export class CustomField<T = any> extends DynamicField<T> {
```
Comment (igortg) [inline src/ng-xform/fields/custom-field.ts:4]: I'd prefer TemplateField. What's the opinion of @lincrosenbach @lvoliveira?

**Review by igortg — CHANGES_REQUESTED** [review]

File: src/ng-xform/fields/custom-field.ts (hunk 2 above)
Comment (edetec) [reply src/ng-xform/fields/custom-field.ts:4]: I think this field is composed of a formControl + a template and is motivated to give the programmer a way to add a custom field to the form, right?

Comment (lvoliveira) [reply src/ng-xform/fields/custom-field.ts:4]: I didn't like TemplateField either, looks like it is a Template of a field

Comment (lincrosenbach) [reply src/ng-xform/fields/custom-field.ts:4]: Perhaps, we should find other examples with custom/template implementation in shared libraries and follow the most used case.

Comment (lincrosenbach) [reply src/ng-xform/fields/custom-field.ts:4]: Custom and Template are both clear to me what it is.

Comment (igortg) [reply src/ng-xform/fields/custom-field.ts:4]: ok then, let it be

**Review by igortg — APPROVED** [review]

**Review by lvoliveira — APPROVED** [review]

**Pull request merged**